js/client.js

```diff
@@ -368,6 +368,18 @@
 						}
 					}
 					this.send('/autojoin ' + autojoinIds.join(','));
+
+					var battleAutojoin = (Tools.prefs('battleautojoin') || '').split(',');
```

Comment:
Might be more straightforward for the server to send you a list of battles you're currently in, and just make the "rejoin battle" button very prominent.